File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float owed = 0;
    do
    {
        owed = get_float("Change owed : ");
    }
    while (owed <= 0);


    int cents = round(owed * 100);

    int count_25 = 0;
    while (cents >= 25)
    {
        cents -= 25;
        count_25++;
    }

    int count_10 = 0;
    while (cents >= 10)
    {
        cents -= 10;
        count_10++;
    }

    int count_5 = 0;
    while (cents >= 5)
    {
        cents -= 5;
        count_5++;
    }

    int count_1 = cents;
    int count = count_25 + count_10 + count_5 + count_1;
    printf("25 * %i + 10 * %i + 5 * %i + %i\n", count_25, count_10, count_5, count_1);
    printf("%i\n", count);
}

File: 6_Python/pset/cash/cash.py
from cs50 import get_float


def main():
    while True:
        input = get_float("Change owed: ")
        if input >= 0:
            break

    # セントに直す
    change = input * 100

    # おつりに使えるコインの種類
    coins = [25, 10, 5, 1]
    # コインの枚数
    count = 0

    for coin in coins:
        result = calc(change, coin)
        change = result["remain"]
        count += result["count"]

    print(count)


def calc(remain, coin):
    count = 0
    while remain >= coin:
        count += 1
        remain -= coin
    return {"remain": remain, "count": count}


main()
